### ISSUE TITLE
Request: Warn when returning a Task using a disposable resource

### ISSUE THREAD

**Issue by LordMike** [dotnet/roslyn-analyzers#1752]
#### Analyzer package

N/A

#### Package Version

N/A

#### Diagnostic ID

N/A

#### Repro steps

```csharp
public Task<object> MyAsyncMethod(Stream someStream)
{
    using (var fileStream = File.OpenRead())
    {
        return SomeOtherAsyncTask(fileStream);
     }
}
```

So. In this case, we're using a disposable resource in an async method, but since our method returns a Task too, we don't get any compiler error and therefore might not notice that we should've await'ed the `SomeOtherAsyncTask`.

This particular issue has arisen twice today in our source control, as we've been moving toward Async-methods everywhere. In multiple places, we missed the need to input await's, and simply returned the Tasks (because there's no need to await in certain cases)..

An Analyzer that detects this, would be awesome.

#### Expected behavior

Warning / Error on the async call with a disposable resource as an argument.

#### Actual behavior

No warnings.

**Comment by JohanLarsson** [comment]
Until it is added you can use: https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP013.md

**Comment by sharwell** [comment]
💭 I'm not sure if we're migrating towards Microsoft/vs-threading as the home for asynchronous analyzers, but if so it seems like this would fit in well there.

**Comment by mavasani** [comment]
Agree with @sharwell - please move this request to https://github.com/microsoft/vs-threading
